### DISCUSSION TITLE
Default tab width

### DISCUSSION THREAD

**Discussion by kj** [helix-editor/helix#4105] [Q&A]
Is there any way to change the default tab width for files that don't match an existing language, and the scratch buffer? I want to set tab width to 8 by default to be more in line with the rest of my environment. Is there a default/fallback language I can target in languages.toml, or is this option simply missing?

The closest I could get was to define a new language, but then I need to list out every file I want to use it with:

```
[[language]]
name = 'plain'
scope = 'text.plain'
file-types = ['bookmarks', 'txt']
roots = []
indent = { tab-width = 8, unit = '\t' }
```

**Comment by Omnikar** [comment]
I think you can add configuration to the `text` language.

**Comment by kj** [comment]
Thanks for the suggestion. I tried adding this to languages.toml:

```
[[language]]
name = 'text'
indent = { tab-width = 8, unit = '\t' }

```

But it doesn't seem to recognise the language when I run the editor:

```
Bad language config: missing field `scope` for key `language`
```

**Reply by Omnikar** [reply]
Oh, it looks like Helix's embedded default languages.toml doesn't have the required default configuration for `text`, which makes sense, since it does not have a tree-sitter grammar. I'm not sure exactly what the best way to go about solving this is.

**Reply by kj** [reply]
Do you know what the editor considers the 'language' to be when you open a scratch buffer or a file which is not otherwise recognised? Personally I just want to be able to configure things like tab width for the default state like I would for any supported language.
